tiledb/sm/serialization/query.cc

```diff
@@ -449,6 +452,12 @@ Status reader_from_capnp(
   RETURN_NOT_OK(reader->set_offsets_extra_element(
       reader_reader.getVarOffsetsAddExtraElement()));
 
+  // Offsets mode
+  if (reader_reader.hasVarOffsetsBitsize()) {
```

Comment:
You have the right idea here, but capnp does not generate `hasX` functions for primitive types (and `varOffsetsBitsize` is an `Int32`). Instead, we'll have to determine this by checking against the default value of 0.

So I think `if (reader_reader.getVarOffsetsBitsize() > 0) {` would be appropriate here.